lib/modules/platform/gerrit/index.ts

```diff
@@ -0,0 +1,543 @@
+  labelMappings: {
+    stabilityDaysLabel: 'Renovate-Stability',
+  },
+};
+
+export function writeToConfig(newConfig: typeof config): void {
+  config = { ...config, ...newConfig };
+}
+
+export function initPlatform({
+  endpoint,
+  username,
+  password,
+  gerritLabelMapping,
+}: PlatformParams & RepoGlobalConfig): Promise<PlatformResult> {
+  logger.debug(`initPlatform(${endpoint!}, ${username!})`);
+  if (!endpoint) {
+    throw new Error('Init: You must configure a Gerrit Server endpoint');
+  }
+  if (!(username && password)) {
+    throw new Error(
+      'Init: You must configure a Gerrit Server username/password'
+    );
+  }
+  config.labelMappings = gerritLabelMapping;
+  config.gerritUsername = username;
+  defaults.endpoint = ensureTrailingSlash(endpoint);
+  setBaseUrl(defaults.endpoint);
+  const platformConfig: PlatformResult = {
+    endpoint: defaults.endpoint,
+  };
+  return Promise.resolve(platformConfig);
+}
+
+/**
+ * Get all state="ACTIVE" and type="CODE" repositories from gerrit
+ */
+export async function getRepos(): Promise<string[]> {
+  logger.debug(`getRepos()`);
+  return await client.getRepos();
+}
+
+/**
+ * Clone repository to local directory
+ * @param config
+ */
+export async function initRepo({
+  repository,
+  gitUrl,
+}: RepoParams): Promise<RepoResult> {
+  logger.debug(`initRepo(${repository}, ${gitUrl!})`);
+  const projectInfo = await client.getProjectInfo(repository);
+  const branchInfo = await client.getBranchInfo(repository);
+
+  config = {
+    ...config,
+    repository,
+    head: branchInfo.revision,
+    config: projectInfo,
+    labels: projectInfo.labels ?? {},
+  };
+  const baseUrl = defaults.endpoint!;
+  const url = getGerritRepoUrl(repository, baseUrl);
+  configureScm(repository, config.gerritUsername!);
+  await git.initRepo({ url });
+
+  //abandon "open" and "rejected" changes at startup
+  const rejectedChanges = await findOwnPr({
+    branchName: '',
+    state: 'open',
+    label: '-2',
+  });
+  for (const change of rejectedChanges) {
+    await client.abandonChange(change._number);
+  }
+  const repoConfig: RepoResult = {
+    defaultBranch: config.head!,
+    isFork: false,
+    repoFingerprint: repoFingerprint(repository, baseUrl),
+  };
+  return repoConfig;
+}
+
+async function findOwnPr(
+  findPRConfig: GerritFindPRConfig,
+  refreshCache?: boolean
+): Promise<GerritChange[]> {
+  const filterState = mapPrStateToGerritFilter(findPRConfig.state);
+  const filter = ['owner:self', 'project:' + config.repository!, filterState];
+  if (findPRConfig.branchName !== '') {
+    filter.push(`hashtag:sourceBranch-${findPRConfig.branchName}`);
+  }
+  if (findPRConfig.targetBranch) {
+    filter.push(`branch:${findPRConfig.targetBranch}`);
+  }
+  if (findPRConfig.label) {
+    filter.push(`label:Code-Review=${findPRConfig.label}`);
+  }
+  const changes = await client.findChanges(filter, refreshCache);
+  logger.debug(`findOwnPr(${filter.join(', ')}) => ${changes.length}`);
```

Comment:
```suggestion
  logger.trace(`findOwnPr(${filter.join(', ')}) => ${changes.length}`);
```

It's probybly logged to often on debug level.